src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/tools/GrapplingHook.java

```diff
@@ -22,19 +22,22 @@
 
 public class GrapplingHook extends SimpleSlimefunItem<ItemUseHandler> {
 
+    private final ItemSetting<Boolean> stayininvonuse = new ItemSetting<>("stay-in-inv-on-use", false);
     private final ItemSetting<Integer> despawnTicks = new ItemSetting<>("despawn-seconds", 60);
 
     public GrapplingHook(Category category, SlimefunItemStack item, RecipeType recipeType, ItemStack[] recipe) {
         super(category, item, recipeType, recipe);
 
         addItemSetting(despawnTicks);
+        addItemSetting(stayininvonuse);
     }
 
     @Override
     public ItemUseHandler getItemHandler() {
         return e -> {
             Player p = e.getPlayer();
             UUID uuid = p.getUniqueId();
+            Boolean keepininv = stayininvonuse.getValue();
```

Comment:
Please don't use the boxed `Boolean` class, use the primitive `boolean` instead.